crates/shared/src/event_handling.rs

```diff
@@ -1,18 +1,20 @@
 use crate::{current_block::BlockRetrieving, maintenance::Maintaining};
-use anyhow::{Context, Error, Result};
+use anyhow::{ensure, Context, Error, Result};
 use ethcontract::contract::{AllEventsBuilder, ParseLog};
 use ethcontract::errors::ExecutionError;
-use ethcontract::{
-    dyns::DynTransport, BlockNumber as Web3BlockNumber, Event as EthcontractEvent, EventMetadata,
-};
-use futures::{Stream, StreamExt, TryStreamExt};
+use ethcontract::H256;
+use ethcontract::{dyns::DynTransport, Event as EthcontractEvent, EventMetadata};
+use futures::{future, Stream, StreamExt, TryStreamExt};
 use std::ops::RangeInclusive;
 use tokio::sync::Mutex;
 
 // We expect that there is never a reorg that changes more than the last n blocks.
 pub const MAX_REORG_BLOCK_COUNT: u64 = 25;
 // Saving events, we process at most this many at a time.
 const INSERT_EVENT_BATCH_SIZE: usize = 10_000;
+const MAX_BLOCKS_QUERIED: u64 = 2 * MAX_REORG_BLOCK_COUNT;
```

Comment:
Nothing against this choice, but I'm not sure I really understand why not just use `MAX_REORG_BLOCK_COUNT`?

Reply:
*Edit*:

Ah, is it to be able to process re-orgs of that "last starting point" (i.e. if the service stops, and there is re-org on the block of the last event it indexed, we want to make sure we catch it with the "query by block-hash" case)?

Is there a reason the "query by block-hash" case works better in case these reorgs happen?

Reply:
I noticed that we occasionally enter `update_events` with more than 1 new block (for example 3 new blocks with no reorg). _Not sure if this happens in POS world though_.

So if we would use `MAX_REORG_BLOCK_COUNT` instead of  `MAX_BLOCKS_QUERIED`, and our full range of blocks being always `MAX_REORG_BLOCK_COUNT+3` blocks long, that would trigger both `history fetch` on 3 blocks and `latest fetch` on 25 blocks (so we would have to **REPLACE 28** blocks).
With using `MAX_BLOCKS_QUERIED` it would evaluate to no `history fetch` and `latest fetch` on `MAX_REORG_BLOCK_COUNT+3` and then algorithm would continue with `detect_reorg_path` that would shorten it to only 3 blocks (it would detect no reorg and common block with `last_handled_blocks`) and then it would evaluate to only **APPEND 3** new blocks which is what we want.
So essentially, we need  `MAX_BLOCKS_QUERIED > MAX_REORG_BLOCK_COUNT`, I was not sure how much so went with 2x.

Reply:
Ah ok - this makes sense. Its because we want to avoid the "query by range" case because its more inefficient.

OK - lets keep it as is then (maybe a short comment would be nice).

Reply:
Added comment.